### ISSUE TITLE
[BUG] APIDataset silently corrupts all labels to zeros when y contains integers instead of "positive"/"negative" strings

### ISSUE THREAD

**Issue by rupeshca007** [gc-os-ai/pyaptamer#426]
In `pyaptamer/datasets/dataclasses/_api.py`, `APIDataset._prepare_data` converts labels at line 102 by checking equality with the string `"positive"`:

    y = torch.tensor((y == "positive").astype(int))   # _api.py:102

If `y` is an integer array (e.g. `np.array([0, 1, 1, 0])`), NumPy compares each integer to the string `"positive"`, which returns `False` for every element. The resulting tensor is all-zeros — every sample is silently labelled as negative — with no error, warning, or assertion.

**To Reproduce**

    import numpy as np
    from pyaptamer.datasets.dataclasses._api import APIDataset

    y_int = np.array([0, 1, 1, 0])
    print((y_int == "positive").astype(int))   # [0, 0, 0, 0]  ← all zeros

Or inspect the line directly:

    grep -n "positive" pyaptamer/datasets/dataclasses/_api.py
    # 102:        y = torch.tensor((y == "positive").astype(int))

**Expected behavior**

Either:

1. Raise a clear `ValueError` when `y` does not consist of `"positive"`/`"negative"` strings, so the user is told what format is required instead of silently receiving corrupted labels; or
2. Accept integer labels (0 / 1) as a second valid input format alongside string labels.

The class docstring already documents that only `"positive"` / `"negative"` strings are supported, so option 1 (a defensive validation guard) is the minimal, surgical fix:

    if not np.isin(y, ["positive", "negative"]).all():
        raise ValueError(
            "Labels `y` must contain only 'positive' or 'negative' strings. "
            f"Got unique values: {np.unique(y).tolist()}"
        )

**Additional context**

    grep -rn "_prepare_data\|y == " pyaptamer/datasets/ --include="*.py"
    # datasets/dataclasses/_api.py:102:        y = torch.tensor((y == "positive").astype(int))

This is the only string-equality label conversion in the datasets tree. A one-line validation guard fully scopes the fix; no API change, no import change, no logic change to the conversion itself.

**Versions**

Reproduced on `main` at commit `3c8b502`.

    pyaptamer == 0.1.0a1

**Comment by siddharth7113** [comment]
Hi,

Did you use AI to find the issue?
We had recently closed a similar issue. #387
